src/FsCodec.NewtonsoftJson/Codec.fs

```diff
@@ -44,10 +44,27 @@ module Core =
                 // TOCONSIDER as noted in the comments on RecyclableMemoryStream.ToArray, ideally we'd be continuing the rental and passing out a Span
                 ms.ToArray()
 
-            member __.Decode(json : byte[]) =
+            member __.Decode(json : byte[]) : 'a =
                 use ms = Utf8BytesEncoder.wrapAsStream json
                 use jsonReader = Utf8BytesEncoder.makeJsonReader ms
-                serializer.Deserialize<'T>(jsonReader)
+                let returnType = typeof<'a>
+                if returnType = typeof<Guid> then
+                    json
+                    |> System.Text.Encoding.ASCII.GetString
+                    |> Guid.Parse
+                    |> unbox
+                elif returnType = typeof<bool> then
+                    json
+                    |> System.Text.Encoding.ASCII.GetString
```

Comment:
While this may be technically correct, I think I'd stick to using UTF8 in the interests of not overfitting?